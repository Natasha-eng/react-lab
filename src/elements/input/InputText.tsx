import { ChangeEvent, FocusEvent } from "react";
import inputStyles from "./inputText.module.css";

interface IInputText {
  name: string;
  type: string;
  label: string;
  value: string;
  error: string;
  onChangeValueHandler: (value: string) => void;
  onBlurHander: (value: string) => void;
}

export default function InputText(props: IInputText): JSX.Element {
  const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
    props.onChangeValueHandler(e.target.value);
  };

  const onBLurInputHandler = (e: FocusEvent<HTMLInputElement>) => {
    props.onBlurHander(e.target.value);
  };
  return (
    <div className={inputStyles.inputWrapper}>
      <div className={inputStyles.input}>
        <label htmlFor={props.name}>{props.label}</label>
        <input
          type={props.type}
          name={props.name}
          value={props.value}
          onChange={onChangeHandler}
          onBlur={onBLurInputHandler}
        />
      </div>
      {props.error && <div className={inputStyles.error}>{props.error}</div>}
    </div>
  );
}
