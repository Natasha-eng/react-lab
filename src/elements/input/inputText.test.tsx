// eslint-disable-next-line no-use-before-define
import React from "react";
import { unmountComponentAtNode } from "react-dom";
import renderer from "react-test-renderer";
import InputText from "./InputText";

const mockData = {
  name: "username",
  type: "text",
  label: "Username",
  value: "natasha",
  onChangeValueHandler: () => {},
  onBlurHander: () => {},
  error: "error",
};

describe("InputText component", () => {
  let container: HTMLDivElement;
  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
  });

  it("should render Input component", () => {
    const tree = renderer
      .create(
        <InputText
          name={mockData.name}
          type={mockData.type}
          label={mockData.label}
          value={mockData.value}
          onChangeValueHandler={mockData.onChangeValueHandler}
          onBlurHander={mockData.onBlurHander}
          error={mockData.error}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
