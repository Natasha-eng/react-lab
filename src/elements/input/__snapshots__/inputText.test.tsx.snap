// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText component should render Input component 1`] = `
<div
  className="inputWrapper"
>
  <div
    className="input"
  >
    <label
      htmlFor="username"
    >
      Username
    </label>
    <input
      name="username"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="natasha"
    />
  </div>
  <div
    className="error"
  >
    error
  </div>
</div>
`;
